test/jdk/javax/swing/JInternalFrame/InternalFrameTitleButtonTest.java

```diff
@@ -52,18 +49,21 @@ public class InternalFrameTitleButtonTest {
     private static JInternalFrame iframe;
 
     public static void main(String[] args) throws Exception {
-        if (OSInfo.getOSType() == OSInfo.OSType.WINDOWS) {
-            UIManager.setLookAndFeel(
-                   "com.sun.java.swing.plaf.windows.WindowsClassicLookAndFeel");
-            try {
-                test();
-            } finally {
-                SwingUtilities.invokeAndWait(() -> {
-                    if (frame != null) {
-                        frame.dispose();
-                    }
-                });
-            }
+        String osName = System.getProperty("os.name");
+        if(!osName.toLowerCase().contains("win")) {
+            System.out.println("The test is applicable only for Windows.");
+            return;
+        }
+        UIManager.setLookAndFeel(
+               "com.sun.java.swing.plaf.windows.WindowsClassicLookAndFeel");
+        try {
```

Comment:
Actually, I meant that you can skip checking for Windows at all: the test has `@requires (os.family == "windows")` tag. Then setting *Windows Classic Look and Feel* will fail on other platforms but Windows, that is `UIManager.setLookAndFeel` will throw an exception which will eventually fail the test.

Whatever way you prefer.